ygen/ir.go

```diff
@@ -338,23 +358,65 @@ const (
 
 // YANGNodeDetails stores the YANG-specific details of a node
 // within the schema.
+// TODO(wenbli): Split this out so that parts can be re-used by
+// ParsedDirectory.
 type YANGNodeDetails struct {
 	// Name is the name of the node from the YANG schema.
 	Name string
 	// Defaults represents the 'default' value(s) directly
 	// specified in the YANG schema.
 	Defaults []string
-	// Module stores the name of the module that instantiates
-	// the node.
-	Module string
-	// Path specifies the absolute YANG schema node path.
+	// BelongingModule is the module in whose namespace the node belongs.
+	BelongingModule string
+	// RootModule is the module in which the YANG tree the node is attached
+	// to was instantiated (rather than the module that has the same
+	// namespace as the node).
+	//
+	// In this example, leaf has a RootModule of "openconfig-simple", while a
+	// BelongingModule of "openconfig-augment". The DefiningModule is
+	// "openconfig-augment".
+	//
+	//   module openconfig-augment {
+	//     import openconfig-simple { prefix "t"; }
+	//     augment "/t:parent/child/state" {
+	//       leaf zero { type string; }
+	//     }
+	//   }
+	RootModule string
+	// DefiningModule is the module that contains the textual definition of
+	// the field.
+	DefiningModule string
+	// Path specifies the absolute YANG schema node path that can be used
+	// to index into the ParsedDirectory map in the IR. It includes the
+	// module name as well as choice/case elements.
 	Path string
+	// SchemaPath specifies the absolute YANG schema node path. It does not
+	// include the module name nor choice/case elements in the YANG file.
+	SchemaPath string
 	// ResolvedPath specifies the leafref-resolved absolute YANG schema
 	// node path.
 	ResolvedPath string
```

Comment:
This field is only populated for `leafref` fields, I just updated the doc comment to make this clear.

Reply:
I'm still not clear what this actually does. Is this the path to which the leafref refers? Can you explain how it's used please?

Reply:
I tried to clarify its meaning here: https://github.com/openconfig/ygot/pull/658, the use case is https://github.com/openconfig/ygot/blob/056010ada41e6b17b3b89dc8427d9f193f937490/ygen/protogen.go#L1179-L1180

Where the leafref path is being used to generate the tag number, which is actually buggy because two leafrefs could point to the same location. This was deliberately kept in order to preserve backwards-compatibility: https://github.com/openconfig/ygot/pull/610#discussion_r781510037

Reply:
https://github.com/openconfig/ygot/pull/658 Addresses the comment here.